pkg/apis/numaflow/v1alpha1/pipeline_types.go

```diff
@@ -175,17 +175,52 @@ func (p Pipeline) GetDownstreamEdges(vertexName string) []Edge {
 	return result
 }
 
+// HasSideInputs returns if the pipeline has side inputs.
+func (p Pipeline) HasSideInputs() bool {
+	return len(p.Spec.SideInputs) > 0
+}
+
 func (p Pipeline) GetDaemonServiceName() string {
 	return fmt.Sprintf("%s-daemon-svc", p.Name)
 }
 
 func (p Pipeline) GetDaemonDeploymentName() string {
 	return fmt.Sprintf("%s-daemon", p.Name)
 }
+
 func (p Pipeline) GetDaemonServiceURL() string {
 	return fmt.Sprintf("%s.%s.svc:%d", p.GetDaemonServiceName(), p.Namespace, DaemonServicePort)
 }
 
+func (p Pipeline) GetSideInputDeploymentName(sideInputName string) string {
+	return fmt.Sprintf("%s-si-%s", p.Name, sideInputName)
+}
+
+func (p Pipeline) GetSideInputsStoreName() string {
+	return p.Name
```

Comment:
Are we setting SideInputs store name the same as pipeline name? Maybe it worth documenting the rationale behind.